src/containers/table/Table.js

```diff
@@ -108,60 +117,82 @@ class TableView extends React.Component {
-        companies: changeNumOrder(this.state.companies,
-          'funding', 'desc'), order: 'desc'
-      });
+      if (this.state.order === 'desc' || this.state.order === null) {
+        newState = R.merge(newState, {
+          companies: changeNumOrder(this.state.companies,
+            'funding', 'asc'), order: 'asc'
+        });
+      } else {
+        newState = R.merge(newState, {
+          companies: changeNumOrder(this.state.companies,
+            'funding', 'desc'), order: 'desc'
+        });
+      }
     }
     this.setState({companies: newState.companies, sortBy: 'funding', order: newState.order});
   }
 
   handleEmployeesClick(e) {
     let newState = R.clone(this.state);
-    if (this.state.order === 'desc' || this.state.order === null) {
+    if (this.state.sortBy != 'employees') {
       newState = R.merge(newState, {
-        companies: changeNumOrder(this.state.companies,
-          'employees', 'asc'), order: 'asc'
+        companies: changeNumOrder(this.state.companies, 'employees', 'desc'),
+        order: 'desc'
       });
     } else {
-      newState = R.merge(newState, {
-        companies: changeNumOrder(this.state.companies,
-          'employees', 'desc'), order: 'desc'
-      });
+      if (this.state.order === 'desc' || this.state.order === null) {
+        newState = R.merge(newState, {
+          companies: changeNumOrder(this.state.companies,
+            'employees', 'asc'), order: 'asc'
+        });
+      } else {
+        newState = R.merge(newState, {
+          companies: changeNumOrder(this.state.companies,
+            'employees', 'desc'), order: 'desc'
+        });
+      }
     }
     this.setState({companies: newState.companies, sortBy: 'employees', order: newState.order});
   }
 
   handleFoundedClick(e) {
     let newState = R.clone(this.state);
-    if (this.state.order === 'desc' || this.state.order === null) {
+    if (this.state.sortBy != 'foundedOn') {
       newState = R.merge(newState, {
-        companies: changeNumOrder(this.state.companies,
-          'foundedOn', 'asc'), order: 'asc'
+        companies: changeNumOrder(this.state.companies, 'foundedOn', 'desc'),
+        order: 'desc'
       });
     } else {
-      newState = R.merge(newState, {
-        companies: changeNumOrder(this.state.companies,
-          'foundedOn', 'desc'), order: 'desc'
-      });
+      if (this.state.order === 'desc' || this.state.order === null) {
+        newState = R.merge(newState, {
+          companies: changeNumOrder(this.state.companies,
+            'foundedOn', 'asc'), order: 'asc'
+        });
+      } else {
+        newState = R.merge(newState, {
+          companies: changeNumOrder(this.state.companies,
+            'foundedOn', 'desc'), order: 'desc'
+        });
+      }
     }
-    this.setState({companies: newState.companies, sortBy: 'funding', order: newState.order});
+    this.setState({companies: newState.companies, sortBy: 'foundedOn', order: newState.order});
```

Comment:
do you have any ideas, how we could write those clickHandlers in more abstaract way that we could use only one handler for all functions ?